### PR TITLE
Convert DateTimePicker tests to WDIO

### LINKED ISSUE #1302: Replace Nightwatch Tests with Webdriver.io Tests
# Issue Description
Remove the existing nightwatch tests and replace them with webdriver.io tests. This has been done for some components but these need to happen for the remaining components:

Checked off means done or claimed

- [x] terra-content-container
- [x] terra-demographics-banner
- [x] terra-divider
- [x] terra-dynamic-grid
- [x] terra-form-checkbox
- [x] terra-form-field
- [x] terra-form-textarea
- [x] terra-form-select
- [x] terra-heading
- ~terra-hookshot~ Migrated to terra-framework https://github.com/cerner/terra-framework/pull/128
- [x] terra-icon
- [x] terra-image
- [x] terra-profile-image
- [x] terra-progress-bar
- [x] terra-props-table
- ~terra-slide-panel~ Migrated to terra-framework https://github.com/cerner/terra-framework/pull/131
- [x] terra-spacer
- [x] terra-status
- [x] terra-text
- [x] terra-toggle
- [x] terra-grid
- [x] terra-list
- [x] terra-responsive-element
- [x] terra-signature
- [x] terra-slide-group
- [x] terra-table
- [x] terra-toggle-button
- [x] terra-abstract-modal
- [x] terra-date-picker
- [x] terra-menu
- [x] terra-search-field

## Remaining Components
- [x] terra-data-time-picker
- [x] terra-embedded-content-consumer

## Issue Type

- [ ] New Feature
- [ ] Enhancement
- [ ] Bug
- [x] Other

### PR DESCRIPTION
**Pull request by JakeLaCombe**
### Summary
Resolves #1302 

In addition to DateTimePicker, I made a few other cleanup changes across the board.

### CODE REVIEW

**Comment by JakeLaCombe** [comment]
Can't just yet. The terra-embedded-content-container tests haven't been converted just yet.

However, if we don't care about running the nightwatch tests against just one package, I say we can remove all those. @bjankord @ryanthemanuel @mjhenkes thoughts?

**Comment by bjankord** [comment]
Let's remove the nightwatch deps and config in a separate PR.

**Comment by emilyrohrbough** [comment]
Yeah I jumped the gun and forgot the others were still in review/work in progress.

File: packages/terra-date-time-picker/tests/wdio/date-time-picker-spec.js [hunk 1]
```diff
@@ -0,0 +1,337 @@
...
+      before(() => {
+        browser.url('/#/raw/tests/terra-date-time-picker/date-time-picker/date-time-picker-dst');
+        browser.click('input[name="terra-time-hour-input"]');
+        browser.click('input[name="terra-time-minute-input"]');
+        browser.waitForVisible('[class*="time-clarification"]', 1000);
+      });
+
+      Terra.should.matchScreenshot('0', { viewports, selector: '[class*="abstract-modal"]' });
+
+      it('hids the offset button', () => {
```
Comment (tomleewu) [inline packages/terra-date-time-picker/tests/wdio/date-time-picker-spec.js]: nit: _hids_ -> hides

File: packages/terra-date-time-picker/tests/wdio/date-time-picker-spec.js [hunk 2]
```diff
@@ -0,0 +1,337 @@
...
```
Comment (tomleewu) [inline packages/terra-date-time-picker/tests/wdio/date-time-picker-spec.js]: nit: _hids_ -> hides

**Comment by bjankord** [comment]
Looks like there may be some screenshots that need updated. The last build failed with the following errors:

<details not-open>
<summary>Error messages</summary>


<pre><code>
1) DateTimePicker Launches and Dismisses modal manager after date selection [0] to be within the mismatch tolerance:
expected to be within the mismatch tolerance, but received the following comparison results 
{
  "viewport": "tiny",
  "misMatchPercentage": 0.97
},{
  "viewport": "large",
  "misMatchPercentage": 0.14
}
running chrome
AssertionError: expected to be within the mismatch tolerance, but received the following comparison results 
{
  "viewport": "tiny",
  "misMatchPercentage": 0.97
},{
  "viewport": "large",
  "misMatchPercentage": 0.14
}



1) Form-Textarea AutoResizable Textarea Blocks autoresizing for when DocumentTouch exists and browser screen is not desktop size [1] to be within the mismatch tolerance:
expected to be within the mismatch tolerance, but received the following comparison results 
{
  "viewport": "tiny",
  "misMatchPercentage": 0.07
},{
  "viewport": "large",
  "misMatchPercentage": 0
}
running chrome
AssertionError: expected to be within the mismatch tolerance, but received the following comparison results 
{
  "viewport": "tiny",
  "misMatchPercentage": 0.07
},{
  "viewport": "large",
  "misMatchPercentage": 0
}
</code></pre>
</details>

**Comment by JakeLaCombe** [comment]
I found it difficult to run screenshots against tests that utilize the current date for the datepicker. For this, on the one test we click on the today button, I switched it to be a date in 2017, and click on another date that isn't today. The test was just testing to ensure that the modal opens and closes when a date is switched, so I didn't see an issue in switching the test date.

**Pull request merged**